Inventory

jaewon-ma opened the pull request:
There are two main methods inside Inventory. add() lets you add an item to your inventory, while remove() removes an item. Items can stack, and adding and removing only work in increments of 1.

jaewon-ma committed:
WIP

jaewon-ma committed:
Update Inventory.java

jaewon-ma committed:
WIP

jaewon-ma committed:
WIP

HN67 commented on src/jerra/item/Inventory.java:
```diff
@@ -1,4 +1,43 @@
+
+    public void add(Item item) {
+        if (inventory.size() > 0) {  
+            for (Item i : inventory.keySet()) {
```
inventory.containsKey(item) would probably remove the need for this loop

HN67 commented:
We could also add a method that would consume another Inventory, maybe add a method that takes an item and a quantity so that we can add/remove multiple at a time. Do we want to merge just inventory and then work on loot, or do loot in this branch as well?

HN67 committed:
Cleanup inventory and fix item hashing

rupertraphael committed:
Merge branch 'master' into Inventory

rupertraphael committed:
Refactor .equals for Item and add tests.

rupertraphael committed:
Edit ItemTest class comment.

rupertraphael commented:
Just added tests for Item.

HN67 committed:
Add support for consuming and clearing inventories

HN67 committed:
Merge branch 'Inventory' of https://github.com/HN67/Jerra into Inventory

HN67 committed:
Update Loot to modern architecture

HN67 committed:
Pack collision logic into helper class with generic method

HN67 committed:
Package interaction logic into Collider

Uses generic method, will allow for other types of interaction

HN67 committed:
Add Loot tracking capability to Room

HN67 committed:
Implement Loot interface into Player and Lootbag

Lootbag dissapears when it is picked up

HN67 committed:
Improve Inventory to implement Copyable and self return on add methods

jaewon-ma committed:
Created Medkit/AR15

Created the Medkit (healing item) and AR15 (new weapon) items, Medkit is completed(for the most part) and AR15 is incompleted.

HN67 committed:
Integreate Loot fully into the Room, also create Visual interface

HN67 committed:
Merge branch 'Inventory' of https://github.com/HN67/Jerra into Inventory

HN67 committed:
Comment out compile errors temporarily in AR15

HN67 committed:
Merge branch 'master' into Inventory

HN67 committed:
Resolve conflicts
